downloader.go

```diff
@@ -20,16 +21,22 @@ const (
 	IDLE_TIMEOUT          = 10 * time.Second
 	RETRY_WAIT_MIN        = 500 * time.Millisecond
 	RETRY_WAIT_MAX        = 5 * time.Second
+	MAX_JITTER            = 200 * time.Millisecond
 	NoBytesReceived       = -1
 )
 
-func Backoff(min, max time.Duration, attemptNum int) time.Duration {
-	mult := math.Pow(2, float64(attemptNum)) * float64(min)
+func backoff(minRetry, maxRetry, maxJitter time.Duration, attemptNum int) time.Duration {
+	mult := math.Pow(2, float64(attemptNum)) * float64(minRetry)
 	sleep := time.Duration(mult)
-	if float64(sleep) != mult || sleep > max {
-		sleep = max
+	if float64(sleep) != mult || sleep > maxRetry {
+		sleep = maxRetry
 	}
-	return sleep
+	return sleep + jitter()
+}
+
+func jitter() time.Duration {
+	rand.Seed(time.Now().UnixNano())
```

Comment:
@prycey77 IIRC, you only need to seed the random library once - usually in an `init()` function. It also looks like this function call claims it isn't safe for concurrent use. Can you move this up part to an `init()` function?